frontend/src/metabase/lib/api.js

```diff
@@ -0,0 +1,83 @@
+/* @flow */
+
+import querystring from "querystring";
+
+import EventEmitter from "events";
+
+let events = new EventEmitter();
+
+function makeMethod(method: string, hasBody: boolean = false) {
```

Comment:
It might be nice to have a couple quick unit tests for this function since it looks like it's going to be one of the main interfaces to the API. Down to do that in a follow up if you'd prefer.